### PR TITLE
OpenAPI V3 Enable Beta

### PR DESCRIPTION
**Pull request by Jefftree**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:

Rebases off two approved PRs:
- https://github.com/kubernetes/kubernetes/pull/109010
- https://github.com/kubernetes/kubernetes/pull/109079

Only the latest commit is for this PR

Due to the amount of time it's taking to merge #109010 (a lot of conflicts in swagger.json/openapiv3*.json, I'd like to combine these into one PR and just rebase this for merge.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
OpenAPI V3 is turned on by default
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/features/kube_features.go [hunk 1]
```diff
@@ -1009,7 +1009,7 @@ var defaultKubernetesFeatureGates = map[featuregate.Feature]featuregate.FeatureS
 	genericfeatures.APIPriorityAndFairness:              {Default: true, PreRelease: featuregate.Beta},
 	genericfeatures.OpenAPIEnums:                        {Default: true, PreRelease: featuregate.Beta},
 	genericfeatures.CustomResourceValidationExpressions: {Default: false, PreRelease: featuregate.Alpha},
-	genericfeatures.OpenAPIV3:                           {Default: false, PreRelease: featuregate.Alpha},
+	genericfeatures.OpenAPIV3:                           {Default: true, PreRelease: featuregate.Beta},
```
Comment (lavalamp) [inline pkg/features/kube_features.go:1012]: I don't think we are defaulting beta features on anymore?

Comment (lavalamp) [reply pkg/features/kube_features.go:1012]: Although [Jordan seems to think the rule doesn't apply to this](https://kubernetes.slack.com/archives/C0EG7JC6T/p1647964781225229?thread_ts=1647954978.154789&cid=C0EG7JC6T), I'm not so sure -- e.g. are you 100% sure you're not going to change the url structure for GA?o

Comment (Jefftree) [reply pkg/features/kube_features.go:1020]: /cc @apelisse 

We're planning on keeping the same interface for GA

Comment (lavalamp) [reply pkg/features/kube_features.go:1020]: It's gotta be more than just a plan! Imagine a v1.25 client using openapi v3 by default which happens to talk to a 1.24 cluster--

Comment (lavalamp) [reply pkg/features/kube_features.go:1020]: OK, that's good enough for me!

**Comment by Jefftree** [comment]
Rebased

**Comment by helayoty** [comment]
@Jefftree  👋 I'm in the release 1.24 bug triage shadow. Would you please update the priority field for us to track it better?

**Comment by Jefftree** [comment]
/test pull-kubernetes-unit 
failure should be unrelated

**Comment by Jefftree** [comment]
/retest
Looks like there's an issue for the flake already

**Pull request merged**